`rhel-edge-commit-rt` format definition breaks image test case generation on non-`x86_64` architectures

The `tools/test-case-generators/format-request-map.json` defines `rhel-edge-commit-rt` request as `rhel-edge-commit` image type:
https://github.com/osbuild/osbuild-composer/blob/cad7f7ff63fd6d1e09dcb33aa2a02ad69add1059/tools/test-case-generators/format-request-map.json#L35-L39

`tools/test-case-generators/generate-test-cases` filters the list of format requests using the `"image-type"` field:
https://github.com/osbuild/osbuild-composer/blob/cad7f7ff63fd6d1e09dcb33aa2a02ad69add1059/tools/test-case-generators/generate-test-cases#L223-L224

This makes the `rhel-edge-commit-rt` to be generated every time when one specifies the image type `rhel-edge-commit`. While this is not an issue for x86_64 architecture, this makes the `generate-test-cases` always fail on any other architecture for RHEL distro, because the RT repos are defined only for x86_64:
https://github.com/osbuild/osbuild-composer/blob/cad7f7ff63fd6d1e09dcb33aa2a02ad69add1059/tools/test-case-generators/repos.json#L93-L97

The produced error is:
```
generating test case for rhel-edge-commit-rt
No match for argument: kernel-rt
panic: Could not depsolve: DNF error occured: MarkingErrors: Error occurred when marking packages for installation: Problems in request:
missing packages: kernel-rt

goroutine 1 [running]:
main.main()
        /root/osbuild-composer/cmd/osbuild-pipeline/main.go:120 +0xfa8
exit status 2
```

It would probably make more sense to filter the list of generated image test cases in `tools/test-case-generators/generate-test-cases` using `output_format`, instead of `filtered_request["compose-request"]["image-type"]`

--- a/tools/test-case-generators/format-request-map.json
+++ b/tools/test-case-generators/format-request-map.json
@@ -69,7 +69,8 @@
         }
       }
     },
-    "overrides": {}
+    "overrides": {},
+    "supported_arches": ["x86_64"]
   },
   "rhel-edge-commit": {
     "compose-request": {
@@ -108,7 +109,8 @@
         }
       }
     },
-    "overrides": {}
+    "overrides": {},
+    "supported_arches": ["x86_64"]
   },
   "fedora-iot-commit": {
     "compose-request": {

--- a/tools/test-case-generators/generate-test-cases
+++ b/tools/test-case-generators/generate-test-cases
@@ -242,6 +242,11 @@ def main(distro, arch, image_types, keep_image_info, store, output, with_customi
         if filtered_request["compose-request"]["image-type"] not in image_types:
                 continue
         filtered_request["compose-request"]["distro"] = distro
+        # if the compose-request has specified supported arches, then generate
+        # the test case only if the requested arch is in the list
+        supported_arches = filtered_request.get("supported_arches")
+        if supported_arches is not None and arch not in supported_arches:
+            continue
         filtered_request["compose-request"]["arch"] = arch
         filtered_request["compose-request"]["repositories"] = repos_dict[distro][arch]
 

